classes/UpgradeSelfCheck.php

```diff
@@ -465,4 +480,64 @@ public function runPrestaShopCoreChecks()
 
         return true;
     }
+
+    /**
+     * Get php min and max compatible versions for target prestashop version
+     *
+     * @return array
+     */
+    public function phpCompatibleVersions()
+    {
+        $previousShopVersion = false;
+        $length = count(self::PHP_PS_VERSIONS);
+        $i = 0;
+        $currentShopVersion = null;
+
+        foreach (self::PHP_PS_VERSIONS as $prestashopVersion => $phpVersions) {
```

Comment:
This loop seems a bit complex to find the matching version 🤔
Couldn't it be simplified this way:
```php
$currentShopVersionIndex = 0;
foreach (self::PHP_PS_VERSIONS as $prestashopVersion => $phpVersions) {
    // We reached a version higher than upgrade version, we get out of the loop
    if (version_compare($this->upgrader->version_num, $prestashopVersion, '<')) {
        break;
    }
    currentShopVersionIndex++;

    return self::PHP_PS_VERSIONS[$currentShopVersionIndex];
}
```

Reply:
I wrote it in Github without actually testing so there may be some flaws in my code, but that's the generic idea And since you implemented unit tests at least it's easy to see if this is wrong 😅

Reply:
It doesn't work unfortunately, I tried it out of good faith with the unit tests, but even after fixing this code, I get the PHP versions for the prestashop version just after the one I actually need (I explained the logic in a comment)